domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -341,139 +341,291 @@ pub mod pallet {
                 time,
-                weak_subjectivity_period_check
+                max_weak_subjectivity_period
             );
 
-            if time > weak_subjectivity_period_check {
+            if time > max_weak_subjectivity_period {
                 log::info!(target: "ethereum-beacon-client","💫 Weak subjectivity period exceeded, blocking bridge.",);
                 <Blocked<T>>::set(true);
                 return Err(Error::<T>::BridgeBlocked.into());
             }
 
-            let sync_committee_bits =
-                get_sync_committee_bits(update.sync_aggregate.sync_committee_bits.clone())
-                    .map_err(|_| Error::<T>::InvalidSyncCommitteeBits)?;
-            Self::sync_committee_participation_is_supermajority(sync_committee_bits.clone())?;
-
-            let block_root: H256 =
-                merkleization::hash_tree_root_beacon_header(update.finalized_header.clone())
-                    .map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?
-                    .into();
-            Self::verify_header(
-                block_root,
-                update.finality_branch,
-                update.attested_header.state_root,
-                config::FINALIZED_ROOT_DEPTH,
-                config::FINALIZED_ROOT_INDEX,
-            )?;
-
-            let current_period = Self::compute_current_sync_period(update.attested_header.slot);
-            let sync_committee = Self::get_sync_committee_for_period(current_period)?;
+            let validators_root = <GenesisValidatorsRoot<T>>::get();
+            Self::validate_light_client_update(&update, update.signature_slot, validators_root)?;
+            let sync_committee_bits = update.sync_aggregate.sync_committee_bits.clone();
 
-            let validators_root = <ValidatorsRoot<T>>::get();
-            Self::verify_signed_header(
-                sync_committee_bits,
-                update.sync_aggregate.sync_committee_signature,
-                sync_committee.pubkeys,
-                update.attested_header,
-                validators_root,
-                update.signature_slot,
-            )?;
-
-            Self::store_finalized_header(block_root, update.finalized_header);
+            let stored_latest_finalized_header_state = <LatestFinalizedHeaderState<T>>::get();
+            let store_period = Self::compute_sync_committee_period(
+                stored_latest_finalized_header_state.beacon_slot,
+            );
+            let update_has_finalized_next_sync_committee =
+                !Self::is_next_sync_committee_known(store_period)
+                    && Self::is_sync_committee_update(&update)
+                    && Self::is_finality_update(&update)
+                    && (Self::compute_sync_committee_period(update.finalized_header.slot)
+                        == Self::compute_sync_committee_period(update.attested_header.slot));
+
+            if (Self::get_sync_committee_sum(sync_committee_bits.to_vec()) * 3
```

Comment:
So the threshold (2/3) is a constant in the ethereum spec? A commnet or a smaller helper function could be likely better than directly using the magic number.

Reply:
Yes. It is constant in the spec.

Reply:
I have left comment there. And for efficiency purpose removed the check as it was done by validate light client as well.